### DISCUSSION TITLE
"g.font was nullptr" flurry of errors with opengl and glfw

### DISCUSSION THREAD

**Discussion by jasonrdunne** [ocornut/imgui#6499] [New Users Build/Link/Run issues ONLY!]
This is the jist of my code, it's literally copy pasted from the example:

	#include "imgui/imgui.h"
	#include "imgui/imgui_impl_glfw.h"
	#include "imgui/imgui_impl_opengl3.h"

	//create and init window
	GLFWwindow* window = glfwCreateWindow(SCREEN_W, SCREEN_H, "VoxelEngine", NULL, NULL);
	if (window == NULL)
	{
		printf("Failed to create GLFW window\n");
		glfwTerminate();
		return -1;
	}
	glfwMakeContextCurrent(window);
	glViewport(0, 0, SCREEN_W, SCREEN_H);

	// Setup Dear ImGui context
	IMGUI_CHECKVERSION();
	ImGui::CreateContext();
	ImGuiIO& io = ImGui::GetIO(); (void)io;
	io.ConfigFlags |= ImGuiConfigFlags_NavEnableKeyboard;     // Enable Keyboard Controls

	// Setup Dear ImGui style
	ImGui::StyleColorsDark();

	// Setup Platform/Renderer backends
	ImGui_ImplGlfw_InitForOpenGL(window, true);
	ImGui_ImplOpenGL3_Init("#version 430");

	while (!glfwWindowShouldClose(window))
        {
		ImGui::Begin("Hello, world!");                          // Create a window called "Hello, world!" and append into it.

		ImGui::Text("This is some useful text.");               // Display some text (you can use a format strings too)
		ImGui::End();

		ImGui::Render();
		ImGui_ImplOpenGL3_RenderDrawData(ImGui::GetDrawData());
        }

However, this gives me the error "Exception thrown: read access violation: g.font  was nullptr" even though apparently a font is supposed to be loaded by default. I add this above the main loop:

	io.Fonts->Build();
	io.Fonts->AddFontDefault();
	ImFont* font = io.Fonts->AddFontFromFileTTF("c:\\Windows\\Fonts\\ArialUni.ttf", 18.0f, nullptr, io.Fonts->GetGlyphRangesJapanese());
	ImGui::PushFont(font);

and now it tells me "g.CurrentWindow was nullptr." If i hadn't done io.Fonts->Build(), then it says "atlas was nullptr". I can't find anyone else having this issue online.

**Comment by ocornut** [comment]
You are not calling ImGui::NewFrame() and corresponding backend functions. Refer to examples application in the examples/ folder.
